### PR TITLE
Rewrite validation reports model/controller; add tests

### PR DESCRIPTION
**Pull request by no-reply**
This slightly reworks the `ValidationReports` model, and updates its consumers--adding tests for the functionality along the way.

There are a few places where integration testing/extensive mocking are called out.  These suggest a need for a more extensive rewrite of `ValidationReports` and/or a `ValidationReportBuilder` pattern to free those modules from that internal knowledge.  In the meanwhile, I think this makes everything work. :)

### CODE REVIEW

File: app/controllers/krikri/validation_reports_controller.rb [hunk 1]
```diff
@@ -19,14 +19,19 @@ class ValidationReportsController < CatalogController
 
     def show
```
Comment (anarchivist) [inline app/controllers/krikri/validation_reports_controller.rb]: Can we add a comment here about how this diverges from how `Blacklight::CatalogController` expects `#index` and `#show` to be used?

File: spec/models/validation_report_spec.rb [hunk 2]
```diff
@@ -26,24 +34,53 @@
...
-      expect(result).to eq :result
+      it 'allows pagination' do
+        subject.rows = '1'
+        page_one_docs = subject.find('sourceResource_title').docs
+        subject.page = '2'
+        expect(subject.find('sourceResource_title').docs)
+          .not_to eq page_one_docs
+      end
+
+      it 'multilies default pagination' do
```
Comment (anarchivist) [inline spec/models/validation_report_spec.rb]: (minor) typo: maybe something like `it 'paginates based on a default number of rows'`

Comment (no-reply) [inline spec/models/validation_report_spec.rb]: fixed.

File: app/controllers/krikri/validation_reports_controller.rb (hunk 1 above)
Comment (no-reply) [inline app/controllers/krikri/validation_reports_controller.rb]: Done.

**Comment by anarchivist** [comment]
I'm :+1:, if you want to :squash: or have someone else look at the review.

**Comment by anarchivist** [comment]
:volcano:

**Pull request merged**